Content.Shared/_RMC14/Xenonids/Construction/ModifyHiveLimitsComponent.cs

```diff
@@ -0,0 +1,18 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared._RMC14.Xenonids.Construction;
+
+/// <summary>
+/// When this construct is created, increases the hive's construct limit by some amount.
+/// If it is destroyed the limits are decreased but any old structures stay grandfathered in,
+/// but they cannot be rebuilt if they get destroyed as well.
+/// </summary>
+[RegisterComponent, Access(typeof(ModifyHiveLimitsSystem))]
+public sealed partial class ModifyHiveLimitsComponent : Component
```

Comment:
This would probably be better as the hive core handling the "can this be built" event and it checks the number of existing hive clusters in a query for example. That way you don't need to check prototypes, the hive core checks "does the thing to be built have hive cluster component" and if so "how many are already built", and if the event is not handled (no hive core) then the building is blocked.

Reply:
wouldnt be predicted since a client would need to have all constructs in pvs range, which is why the info is stored on networked hive field

Reply:
At that point you might as well just pvs override the special constructions themselves for xenos

Reply:
@DrSmugLeaf caveat with your idea is you cant raise events on prototypes so it would have to spawn the thing first then raise event then delete it right after which is bad